pkg/cfn/builder/api_test.go

```diff
@@ -202,6 +209,47 @@ var _ = Describe("CloudFormation template builder API", func() {
 		})
 	})
 
+	Describe("NodeGroupAutoScaling", func() {
+		rs := NewNodeGroupResourceSet(&api.ClusterConfig{
+			ClusterName:       clusterName,
+			AvailabilityZones: testAZs,
+			NodeType:          "t2.medium",
+			Region:            "us-west-2",
+			Addons: api.ClusterAddons{
+				WithIAM: api.AddonIAM{
+					AutoScalingGroup: true,
+				},
+			},
+		}, "eksctl-test-123-cluster", 0)
+		rs.AddAllResources()
+
+		template, err := rs.RenderJSON()
+		It("should serialise JSON without errors", func() {
+			Expect(err).ShouldNot(HaveOccurred())
+		})
+		obj := Template{}
+		It("should parse JSON withon errors", func() {
+			err := json.Unmarshal(template, &obj)
+			Expect(err).ShouldNot(HaveOccurred())
+		})
+
+		It("NodeGroup should have correct tags", func() {
+			Expect(len(obj.Resources)).ToNot(Equal(0))
+			Expect(obj.Resources["AutoScalingControl"]).ToNot(BeNil())
+			Expect(len(obj.Resources["AutoScalingControl"].Properties.PolicyDocument.Statement)).To(Equal(1))
+			Expect(obj.Resources["AutoScalingControl"].Properties.PolicyDocument.Statement[0].Effect).To(Equal("Allow"))
+			Expect(obj.Resources["AutoScalingControl"].Properties.PolicyDocument.Statement[0].Resource).To(Equal("*"))
+			Expect(obj.Resources["AutoScalingControl"].Properties.PolicyDocument.Statement[0].Action).To(Equal([]string{
+				"autoscaling:DescribeAutoScalingGroups",
+				"autoscaling:DescribeAutoScalingInstances",
+				"autoscaling:DescribeLaunchConfigurations",
+				"autoscaling:DescribeTags",
+				"autoscaling:SetDesiredCapacity",
+				"autoscaling:TerminateInstanceInAutoScalingGroup",
```

Comment:
I'm curious (mostly for the record), what are the two extra items, as issue #170 only had these 4:
```JSON
            [
                "autoscaling:DescribeAutoScalingGroups",
                "autoscaling:DescribeAutoScalingInstances",
                "autoscaling:SetDesiredCapacity",
                "autoscaling:TerminateInstanceInAutoScalingGroup"
            ]
```

Reply:
DescribeTags is for Cluster-Autoscaler autodiscovery and DescribeLaunchConfigurations is required for scaling up from Desired size of 0.

I took these from:
https://github.com/helm/charts/tree/master/stable/cluster-autoscaler#iam